push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/APNsPushNotificationSender.java

```diff
@@ -209,4 +209,43 @@ public void messageSendFailed(ApnsNotification message, Throwable e) {
         // null if, why ever, there was no cert/passphrase
         return null;
     }
+
+    /**
+     * Configure the Gateway to the Apns servers.
+     * Default gateway and port can be override with respectively :
+     *  - aerogear.apns.push.host
+     *  - aerogear.apns.push.port
+     *
+     * Feedback gateway and port can be override with  respectively :
+     *  - aerogear.apns.feedback.host
+     *  - aerogear.apns.feedback.port
+     * @param iOSVariant
+     * @param builder
+     */
+    private void configureDestinations(iOSVariant iOSVariant, ApnsServiceBuilder builder) {
+        // pick the destination:
+        if (iOSVariant.isProduction()) {
+            builder.withProductionDestination();
+        } else {
+            builder.withSandboxDestination();
+        }
+
+        //Is the gateway host&port provided by a system property ?
+        if(System.getProperty(AEROGEAR_APNS_PUSH_HOST)!= null){
```

Comment:
yeah, and even move that to a Utils. in the static block, just call the
util. Makes it cleaner

On Thu, Jan 22, 2015 at 12:12 PM, Tadeas Kriz <notifications@github.com>
wrote:

> In
> push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/APNsPushNotificationSender.java
> <https://github.com/aerogear/aerogear-unifiedpush-server/pull/470#discussion_r23368736>
> :
>
> > +     * Feedback gateway and port can be override with  respectively :
> > +     *  - aerogear.apns.feedback.host
> > +     *  - aerogear.apns.feedback.port
> > +     * @param iOSVariant
> > +     * @param builder
> > +     */
> > +    private void configureDestinations(iOSVariant iOSVariant, ApnsServiceBuilder builder) {
> > +        // pick the destination:
> > +        if (iOSVariant.isProduction()) {
> > +            builder.withProductionDestination();
> > +        } else {
> > +            builder.withSandboxDestination();
> > +        }
> > +
> > +        //Is the gateway host&port provided by a system property ?
> > +        if(System.getProperty(AEROGEAR_APNS_PUSH_HOST)!= null){
>
> I would suggest creating a method like so:
>
> String tryGetProperty(String key) {
>     try {
>         return System.getProperty(key);
>     } catch (SecurityException e) {
>         logger.error("Could not get value of property `" + key + "` due to SecurityManager. Using null value.");
>         return null;
>     }
> }
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/aerogear/aerogear-unifiedpush-server/pull/470/files#r23368736>
> .
>



-- 
Matthias Wessendorf

blog: http://matthiaswessendorf.wordpress.com/
sessions: http://www.slideshare.net/mwessendorf
twitter: http://twitter.com/mwessendorf